core/domain/certificate_assessment_services.py

```
@@ -279,6 +297,345 @@ def _has_valid_distinct_assignment(
+    if not validation_result['is_valid']:
+        offering_model = (
+            gae_models.CertificateAssessmentOfferingModel.get_by_id(
+                certificate_id
+            )
+        )
+        assert offering_model is not None
+        offering_model.async_status = 'Blocked'
+        offering_model.put()
+        raise CertificateAssessmentAttemptNotReadyException(
+            'Sorry, this assessment isn\'t ready anymore! We\'ve alerted the creator, and in the meantime you can try a different assessment.'
+        )
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)